app/services/statistics.rb

```diff
@@ -1,16 +1,18 @@
     charts = [
-      CountHosts.new(:count_by => :operatingsystem, :title => ("OS Distribution"), :search =>"os_title=~VAL~"),
-      CountHosts.new(:count_by => :architecture, :title => _("Architecture Distribution"), :search => "facts.architecture=~VAL~"),
-      CountHosts.new(:count_by => :environment, :title => _("Environment Distribution"), :search => "environment=~VAL~"),
-      CountHosts.new(:count_by => :hostgroup, :title => _("Host Group Distribution"), :search => "hostgroup=~VAL~"),
-      CountHosts.new(:count_by => :compute_resource, :title => _("Compute Resource Distribution"), :search => "compute_resource=~VAL~"),
-      CountFacts.new(:count_by => :processorcount, :unit => Nn_('%s core', '%s cores'), :title => _("Number of CPUs"), :search => "facts.processorcount=~VAL1~"),
-      CountFacts.new(:count_by => :manufacturer, :title => _("Hardware"), :search => "facts.manufacturer~~VAL~"),
-      CountNumericalFactPair.new(:count_by => :memory, :title => _("Average Memory Usage")),
-      CountNumericalFactPair.new(:count_by => :swap, :title => _("Average Swap Usage")),
-      CountPuppetClasses.new(:id => :puppetclass, :title => _("Class Distribution"), :search => "class=~VAL1~")
+      CountHosts.new(:count_by => :operatingsystem, :title => ("OS Distribution"), :search =>"os_title=~VAL~", :organization_id => org_id, :location_id => loc_id),
+      CountHosts.new(:count_by => :architecture, :title => _("Architecture Distribution"), :search => "facts.architecture=~VAL~", :organization_id => org_id, :location_id => loc_id),
+      CountHosts.new(:count_by => :environment, :title => _("Environment Distribution"), :search => "environment=~VAL~", :organization_id => org_id, :location_id => loc_id),
+      CountHosts.new(:count_by => :hostgroup, :title => _("Host Group Distribution"), :search => "hostgroup=~VAL~", :organization_id => org_id, :location_id => loc_id),
+      CountHosts.new(:count_by => :compute_resource, :title => _("Compute Resource Distribution"), :search => "compute_resource=~VAL~", :organization_id => org_id, :location_id => loc_id),
+      CountFacts.new(:count_by => :processorcount, :unit => Nn_('%s core', '%s cores'), :title => _("Number of CPUs"), :search => "facts.processorcount=~VAL1~", :organization_id => org_id, :location_id => loc_id),
+      CountFacts.new(:count_by => :manufacturer, :title => _("Hardware"), :search => "facts.manufacturer~~VAL~", :organization_id => org_id, :location_id => loc_id),
+      CountNumericalFactPair.new(:count_by => :memory, :title => _("Average Memory Usage"), :organization_id => org_id, :location_id => loc_id),
+      CountNumericalFactPair.new(:count_by => :swap, :title => _("Average Swap Usage"), :organization_id => org_id, :location_id => loc_id),
+      CountPuppetClasses.new(:id => :puppetclass, :title => _("Class Distribution"), :search => "class=~VAL1~", :organization_id => org_id, :location_id => loc_id)
     ]
 
     if SETTINGS[:locations_enabled]
```

Comment:
Shouldn't we also pass these parameters to these two charts? e.g. so i can see distribution of sub-taxonomies or see a distribution of current org in different locations.

Reply:
yeah, makes sense, I've added also loc id for location distribution which will result in 100% chart but that's hopefully the more reasonable than keeping it in any context

Reply:
the fix is part of second commit